lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -378,9 +389,81 @@ Bool PointConditionalDistribution::hasSimplifiedVersion(Distribution & simplifie
     return true;
   }
 
+  // Mixture
+  Mixture *p_mixture = dynamic_cast<Mixture *>(distribution_.getImplementation().get());
+  if (p_mixture)
+  {
+    Collection<Distribution> atoms(p_mixture->getDistributionCollection());
+    const UnsignedInteger atomsNumber = atoms.getSize();
+    Point newWeights(p_mixture->getWeights());
+    Collection<Distribution> newAtoms(atomsNumber);
+    for (UnsignedInteger i = 0; i < atomsNumber; ++i)
+      {
+	newWeights[i] *= atoms[i].getMarginal(conditioningIndices_).computePDF(conditioningValues_);
+	newAtoms[i] = PointConditionalDistribution(atoms[i], conditioningIndices_, conditioningValues_);
+      }
+    simplified = Mixture(newAtoms, newWeights);
+    return true;
+  }
+
+  // Kernel mixture
+  KernelMixture *p_kernel_mixture = dynamic_cast<KernelMixture *>(distribution_.getImplementation().get());
+  if (p_kernel_mixture)
+  {
+    const Distribution kernel(p_kernel_mixture->getKernel());
+    const Point bandwidth(p_kernel_mixture->getBandwidth());
+    const Sample sample(p_kernel_mixture->getInternalSample());
+    const UnsignedInteger sampleSize = sample.getSize();
+    Collection<Distribution> atoms(sampleSize);
+    Point weights(sampleSize, 1.0);
+    const UnsignedInteger dimension = getDimension();
+    const UnsignedInteger conditioningDimension = conditioningIndices_.getSize();
+    for (UnsignedInteger i = 0; i < sampleSize; ++i)
+      {
+	Collection<Distribution> atomComponents(dimension);
+	for (UnsignedInteger j = 0; j < dimension; ++j)
+	  {
+	    const UnsignedInteger newJ = nonConditioningIndices_[j];
+	    const Scalar hJ = bandwidth[newJ];
+	    atomComponents[j] = kernel * hJ + sample(i, newJ);
+	  } // j
+	atoms[i] = JointDistribution(atomComponents);
+	for (UnsignedInteger j = 0; j < conditioningDimension; ++j)
+	  {
+	    const UnsignedInteger newJ = conditioningIndices_[j];
+	    const Scalar hJ = bandwidth[newJ];
+	    const Scalar xJ = conditioningValues_[j];
+	    weights[i] *= kernel.computePDF((xJ - sample(i, newJ)) / hJ) / hJ;
+	  } // j
+      } // i
+    simplified = Mixture(atoms, weights);
+    return true;
+  }
+
+  // Dirichlet
+  Dirichlet *p_dirichlet = dynamic_cast<Dirichlet *>(distribution_.getImplementation().get());
+  if (p_dirichlet)
+  {
+    const Point theta(p_dirichlet->getTheta());
+    Scalar weight = 1.0;
+    for (UnsignedInteger i = 0; i < conditioningValues_.getSize(); ++i)
+      weight -= conditioningValues_[i];
+    const UnsignedInteger dimension = getDimension();
+    Point newTheta(dimension + 1, theta[p_dirichlet->getDimension()]);
+    for (UnsignedInteger i = 0; i < nonConditioningIndices_.getSize(); ++i)
+      {
+	const UnsignedInteger newI = nonConditioningIndices_[i];
+	newTheta[i] =theta[i];
+      }
+    // Unfortunately we don't know how to multiply a multivariate distribution by a scalar
+    // simplified = Distribution(Dirichlet(newTheta)) * weight;
+    // return true;
+    return false;
+  }
+
```

Comment:
we should probably drop the dirichlet case then

Reply:
I hope I will be able to implement a ScaledDistribution tomorrow, to implement the distribution of DX+b where D is a diagonal matrix (aka a Point) and b a Point. It will be useful in many places, the only difficulty is to find the appropriate name!